### PR TITLE
Add demo for Window Management API

### PR DESCRIPTION
**Pull request by chrisdavidmills**
The [Window Management API](https://w3c.github.io/window-management/) was implemented in Chrome 100 (see [ChromeStatus entry](https://chromestatus.com/feature/5252960583942144)); this PR provides a basic demo to demonstrate usage.

### CODE REVIEW

**Review by michaelwasserman — APPROVED** [review]
This is great, thank you very much! I left some comments that I hope are helpful, and I'm happy to circle back with another look or discussion as appropriate.

File: window-management-api/index.js [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+    "url": "https://developer.mozilla.org/en-US/docs/Web/JavaScript"
+  }
+]
+
+const outputElem = document.querySelector("section");
+
+// Array to hold references to the currently open windows
+let windowRefs = [];
+
+// Constants to represent the width and height of the Chrome UI when calculating the window size to open
```
[the remark this answers is not in the document]
Comment (chrisdavidmills) [reply window-management-api/index.js:26]: ;-)

File: window-management-api/index.js [hunk 2]
```diff
@@ -0,0 +1,192 @@
...
+  btn.addEventListener("click", openWindows);
+}
+
+// Functions for creating the windows
+
+function openWindow(left, top, width, height, url) {
+  const windowFeatures = `left=${left},top=${top},width=${width},height=${height}`;
+  const windowRef = window.open(
+    url,
+    Math.random().toString(), // a target is needed for it to open as a separate window rather than a tab
```
[the remark this answers is not in the document]
Comment (chrisdavidmills) [reply window-management-api/index.js:73]: Ah ha, good call. I've updated it in the next commit, and made the comment a bit simpler and more accurate.

File: window-management-api/index.js [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+    let windowWidth = Math.floor((screen1.availWidth - 3 * WINDOW_CHROME_X) / 3);
+    let windowHeight = Math.floor(screen1.availHeight - WINDOW_CHROME_Y);
+
+    // Open the reference windows in thirds across the entire height of the primary screen
+    openWindow(screen1.availLeft,
+      screen1.availTop,
+      windowWidth,
+      windowHeight,
+      sites[1].url);
+    openWindow(windowWidth + screen1.availLeft + WINDOW_CHROME_X,
```
[the remark this answers is not in the document]
Comment (chrisdavidmills) [reply window-management-api/index.js:138]: Changed in next commit

File: window-management-api/index.js [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+      screen1.availTop,
+      windowWidth,
+      windowHeight,
+      sites[2].url);
+    openWindow((windowWidth * 2) + screen1.availLeft + (WINDOW_CHROME_X * 2),
```
[the remark this answers is not in the document]
Comment (chrisdavidmills) [reply window-management-api/index.js:143]: Updated

File: window-management-api/index.js [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+      sites[0].url);
+  }
+
+  // Check whether one of our popup windows has been closed
+  // If so, close them all
+
+  closeMonitor = setInterval(checkWindowClose, 250);
+
+  function checkWindowClose() {
+    windowRefs.forEach(windowRef => {
```
[the remark this answers is not in the document]
Comment (chrisdavidmills) [reply window-management-api/index.js:163]: Updated. Thanks, this is much better.

File: window-management-api/index.js [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+
+  screenDetails.addEventListener("screenschange", () => {
+    // If the new number of screens is different to the old number of screens, report the difference
+    if (screenDetails.screens.length !== noOfScreens) {
+      console.log(
+        `The screen count changed from ${noOfScreens} to ${screenDetails.screens.length}`,
+      );
+    }
+
+    // If the windows are open, close them and then open them again
```
[the remark this answers is not in the document]
Comment (chrisdavidmills) [reply window-management-api/index.js:185]: Yeah, I don't want complexity to get in the way of clearly demonstrating how the API features work.

It took me long enough to come up with this example ;-)

**Review by michaelwasserman — APPROVED** [review]
lgtm fwiw, ty!

File: window-management-api/index.js [hunk 7]
```diff
@@ -0,0 +1,190 @@
...
+  const btn = document.createElement("button");
+  btn.textContent = "Open learning environment"
+  outputElem.appendChild(btn);
+
+  btn.addEventListener("click", openWindows);
+}
+
+// Functions for creating the windows
+
+function openWindow(left, top, width, height, url) {
```
Comment (caugner) [inline window-management-api/index.js:69]: I'm wondering if it wouldn't be better to have a signature like `openWindow(url, { left, top, width, height })`, i.e. an option object to replace the number parameters?

Comment (caugner) [reply window-management-api/index.js:69]: That's fine. Thanks Chris!

**Pull request merged**